mobile/test/common/integration/base_client_integration_test.h

```diff
@@ -61,17 +63,24 @@ class BaseClientIntegrationTest : public BaseIntegrationTest {
   envoyToMobileHeaders(const Http::TestRequestHeaderMapImpl& request_headers);
   Event::ProvisionalDispatcherPtr dispatcher_ = std::make_unique<Event::ProvisionalDispatcher>();
   envoy_http_callbacks bridge_callbacks_;
-  ConditionalInitializer terminal_callback_;
-  callbacks_called cc_{0, 0, 0, 0, 0, 0, 0, "", &terminal_callback_, {}};
+  ConditionalInitializer* terminal_callback_;
+  std::vector<std::unique_ptr<ConditionalInitializer>> multi_terminal_callbacks_;
+  callbacks_called* cc_;
+  std::vector<std::unique_ptr<callbacks_called>> multi_cc_;
```

Comment:
I guess another option would be to use a struct to hold each per engine object, e.g. `std::vector<PerStreamHandle> multi_engines_` and then access things via `multi_engines_[0].stream`

Not a big deal, might not make things clearer